src/SSHLibrary/pythonclient.py

```diff
@@ -139,7 +142,8 @@ def _stat(self, path):
         return SFTPFileInfo('', attributes.st_mode)
 
     def _create_missing_remote_path(self, path):
-        path = path.encode(self._encoding)
+        if is_string(path):
+            path = path.encode(self._encoding)
```

Comment:
Should use `is_unicode` instead of `is_string` here. Encoding byte strings doesn't make sense.